lib/dor/text_extraction/abbyy/split_alto.rb

```diff
@@ -24,31 +27,30 @@ def output_path(filename)
           File.join(File.dirname(alto_path), filename)
         end
 
-        # rubocop:disable Metrics/AbcSize
         def xml_structure(page)
-          namespaces = { 'xsi:schemaLocation': 'http://www.loc.gov/standards/alto/ns-v3# http://www.loc.gov/alto/v3/alto-3-1.xsd' }.merge(doc.collect_namespaces)
           Nokogiri::XML::Builder.new do |xml|
             xml.alto(namespaces) do
-              doc.css('//OCRProcessing').each { |node| xml.parent << node.dup }
-              doc.css('//Styles').each { |node| xml.parent << node.dup }
+              xml.parent << ocr_processing_nodes
+              xml.parent << style_nodes
               xml.parent << page
             end
           end
         end
 
-        def page_filenames
-          @page_filenames ||= doc.css('//sourceImageInformation').children.select(&:element?).map(&:text)
+        def namespaces
+          @namespaces ||= { 'xsi:schemaLocation': 'http://www.loc.gov/standards/alto/ns-v3# http://www.loc.gov/alto/v3/alto-3-1.xsd' }.merge(doc.collect_namespaces)
         end
```

Comment:
caches namespaces in memory to avoid parsing out of the full doc each time (turns out this is pretty slow on a large XML document)

Reply:
This could theoretically be hard coded. It was in the original script. I had it get it from the doc because it seemed safer but it is relatively low risk to assume the namespaces are always going to be the same.

Reply:
yeah ok ... i think it is fine this way for now ...  during my testing, I noticed that `doc.collection_namespaces` took like 20 seconds for a large XML doc, so doing this once per object is acceptable, doing it once per page was bad

if it turns out we still have performance issues we could hardcode too as you suggest